backend_py/src/lights/rpi_pwm_hardware.py

```diff
@@ -0,0 +1,59 @@
+from rpi_hardware_pwm import HardwarePWM, HardwarePWMException
+from .pwm_controller import PWMController
+from typing import Dict
+import logging
+
+class RPiHardwarePWMController(PWMController):
+    NAME = 'Hardware PWM Controller'
+
+    def __init__(self, chip=0, pins=None) -> None:
+        super().__init__()
+
+        self.pwm_supported = True
+
+        if pins is None:
+            pins = {
+                18: 0,
+                19: 1
+            }
+        self.PWM_PINS = pins
+
+        self.pwm_objects: Dict[int, HardwarePWM] = {}
+
+        for pin in self.PWM_PINS.keys():
+            try:
+                self.pwm_objects[pin] = HardwarePWM(pwm_channel=self.PWM_PINS[pin], hz=7812.5, chip=chip)
+            except HardwarePWMException:
+                logging.warning('Hardware PWM is not enabled. Need to add \'dtoverlay=pwm-2chan\' to /boot/config.txt and reboot.')
+                self.pwm_supported = False
+                break
+            self.pwm_objects[pin].start(0)
+
+        # Make sure the objects are not initialized
+        if not self.pwm_supported:
+            self.pwm_objects = {}
+
+    def is_pwm_pin(self, pin: int) -> bool:
+        '''
+        Return true if the pin is supported but will always return false if pwm is not supported entirely
+        '''
+        return pin in self.PWM_PINS.keys() if self.pwm_supported else False
+    
+    def disable_pin(self, pin: int):
```

Comment:
Future planned feature. Is currently in frontend API and just needs backend implementation. I will add documentation for that information. This is because this is an abstract class and needs to implement every function even if it is not ready yet.